---
title: "Analytics 2: Introduction to the command-line for Bioinformatics"
author: "Module Coordinator Mark Dunning"
output: 
  html_document: 
    css: stylesheets/styles.css
editor_options: 
  chunk_output_type: inline
---

![](images/header-logo.png)

![](images/logo-sm.png)

## Course Content


This course offers an introduction to working with Linux. We will describe the Linux environment so that participants can start to utilize command-line tools and feel comfortable using a text-based way of interacting with a computer. We will present a use-case of working with sequencing data, but the same principles can be applied to other types of high-throuhput data.

## Pre-requisites

No prior programming experience is required. Computers will be provided. You can bring your own laptop to the workshop if you wish but it will need a working wi-fi connection and you will have to let the tutor know in advance so a temporary York password can be created for you.

## Learning Outcomes

After this course you should be able to:

- Connect to a Unix / Linux system
- Navigate around a file system by issuing commands; rather than using a Desktop environment
- Move and copy files and directories within the Linux system
- Work with text files
- Run programs from the command-line
- Write short scripts to document an analysis


## Schedule (Provisional)

Below is an outline schedule, the exact timings may change depending on particular topics and questions of interest that may arise. There is more material than will be covered on the day so please feel free to look through in your own time. 

- 09:30 - 10:00 [Introduction and Course Setup](https://sbc.shef.ac.uk/wr_bbsrc_dtp_analytics2_2025/part0.nb.html)
- 10:00 - 11:00
    + [Introduction to the command line](https://datacarpentry.org/shell-genomics/01-introduction/)
    + [Navigating files and directories](https://datacarpentry.github.io/shell-genomics/02-the-filesystem.html)
- 11:00 - 11:30 BREAK
- 11:30 - 12:15
    + [Working wih files and directories](https://datacarpentry.github.io/shell-genomics/03-working-with-files.html)
    + [Redirection](https://datacarpentry.github.io/shell-genomics/04-redirection.html)
    + [Writing scripts and working with data](https://datacarpentry.github.io/shell-genomics/05-writing-scripts.html)
- 12:15 - 13:00 LUNCH
- 13:00 - 14:30 [Obtaining Sequencing data and quality assessment](https://sbc.shef.ac.uk/wr_bbsrc_dtp_analytics2_2025/part2.nb.html))

- 
## Optional Materials

The previous version of this workshop all had a section on using Bioinformatics workflows and pipelines; with a particular example of RNA-seq. You may also be interested in learning about *version control*.

+ [Intro to Workflows, Pipelines and Workflow Managers](part3.nb.html)
+ [Introduction to Version Control, and git - Presented by The Data Science Hub, University of York](https://github.com/fabiano-pais/Analytics2/blob/main/git-workshop.md)


## Unix "cheatsheet"

- [Unix cheatsheet](https://upload.wikimedia.org/wikipedia/commons/7/79/Unix_command_cheatsheet.pdf)

## Course Feedback

Please use this short form to give us feedback on the workshop. This will be used to inform the planning of future White Rose DTP training

- [Feedback form]()
